segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/AbstractInMemoryChunkStorage.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.mocks;
+
+import io.pravega.segmentstore.storage.chunklayer.BaseChunkStorage;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Base class for InMemory mock implementations.
+ * Allows to simulate ChunkStorage with different supported properties.
+ */
+@Slf4j
+abstract public class AbstractInMemoryChunkStorage extends BaseChunkStorage {
```

Comment:
This class does not bring much value.
why do we need it ?

Reply:
This class is extremely useful for testing.
It has shouldSupportAppend and shouldSupportConcat fields which allows me to control behavior. This is used extensively in tests for chunk manager.  For example `ChunkManagerTests::testSimpleScenarioWithNonAppendProvider` or

I have two concrete implementations `InMemoryChunkStorage` that stores all data in memory and `NoOpChunkStorage` which only stores metadata and ignores the actual data.
```
 **
 * Base class for InMemory mock implementations.
 * Allows to simulate ChunkStorage with different supported properties.
 */
@Slf4j
abstract public class AbstractInMemoryChunkStorage extends BaseChunkStorage {
    /**
     * value to return when {@link AbstractInMemoryChunkStorage#supportsTruncation()} is called.
     */
    @Getter
    @Setter
    boolean shouldSupportTruncation = true;

    /**
     * value to return when {@link AbstractInMemoryChunkStorage#supportsAppend()} is called.
     */
    @Getter
    @Setter
    boolean shouldSupportAppend = true;

    /**
     * value to return when {@link AbstractInMemoryChunkStorage#supportsConcat()} is called.
     */
    @Getter
    @Setter
    boolean shouldSupportConcat = false;

    /**
```

Reply:
okay